src/main/java/com/cloudogu/scm/search/IndexSyncer.java

```diff
@@ -65,6 +65,25 @@ public void ensureIndexIsUpToDate(Index<FileContent> index, Repository repositor
     }
   }
 
+  void reindex(Index<FileContent> index, Repository repository) {
+    try (RepositoryService repositoryService = repositoryServiceFactory.create(repository)) {
+      if (isSupported(repositoryService)) {
+        Stopwatch sw = Stopwatch.createStarted();
+        Indexer indexer = indexerFactory.create(index, repositoryService);
+        try {
+            IndexSyncWorker worker = indexSyncWorkerFactory.create(repositoryService, indexer);
+            worker.reIndex();
+        } finally {
+          LOG.debug("re-index operation finished in {}", sw.stop());
+        }
+      } else {
+        LOG.warn("repository {} could not index, because it does not support combined modifications", repository);
+      }
+    } catch (IOException e) {
+      LOG.error("failed to update index or to check if an update is required for repository {}", repository, e);
+    }
+  }
+
```

Comment:
Tests?

Reply:
Maybe we can combine this with `ensureIndexIsUpToDate` to reuse logic and joint testing ? Otherwise we just duplicate test code that internally tests the same methods.

Reply:
I copied and nested the tests